app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/ProductPhotosFragment.java

```diff
@@ -64,72 +55,56 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
-            public void onImageResponse(boolean value, String response) {
+            if (value && response != null) {
 
-                if (value && response != null) {
+                // a json object referring to base json object
+                JSONObject jsonObject = Utils.createJsonObject(response);
 
-                    // a json object referring to base json object
-                    JSONObject jsonObject = Utils.createJsonObject(response);
-
-                    // a json object referring to images
-                    JSONObject images = null;
-                    try {
-                        images = jsonObject.getJSONObject("product").getJSONObject("images");
-                    } catch (JSONException e) {
-                        e.printStackTrace();
-                    }
-
-                    if (images != null) {
+                // a json object referring to images
+                JSONObject images = null;
+                try {
+                    images = jsonObject.getJSONObject("product").getJSONObject("images");
+                } catch (JSONException e) {
+                    Log.w(ProductPhotosFragment.class.getSimpleName(), "can get product / images in jon", e);
+                }
+                if (images != null) {
+                    final JSONArray names = images.names();
+                    if (names != null) {
                         // loop through all the image names and store them in a array list
-                        for (int i = 0; i < images.names().length(); i++) {
-
+                        for (int i = 0; i < names.length(); i++) {
                             try {
                                 // do not include images with contain nutrients,ingredients or other in their names
                                 // as they are duplicate and do not load as well
-                                if (images.names().getString(i).contains("n") ||
-                                        images.names().getString(i).contains("f") ||
-                                        images.names().getString(i).contains("i") ||
-                                        images.names().getString(i).contains("o")) {
+                                final String namesString = names.getString(i);
+                                if (namesString.contains("n") ||
+                                    namesString.contains("f") ||
+                                    namesString.contains("i") ||
+                                    namesString.contains("o")) {
 
                                     continue;
-
                                 }
-                                imageNames.add(images.names().getString(i));
+                                imageNames.add(namesString);
                             } catch (JSONException e) {
-                                e.printStackTrace();
+                                Log.w(ProductPhotosFragment.class.getSimpleName(), "can get product / images in jon", e);
```

Comment:
`Log.w(ProductPhotosFragment.class.getSimpleName(), "can get product / images in json", e);`